Create olmCatalogDir if doesn't exists in deploy/

darkowlzz opened the pull request:
If olm-catalog/ is removed from deploy/ dir, olm-catalog generation
fails. This change creates deploy/olm-catalog/ dir if it doesn't exist
and avoids failure.

Fixes #495

shawn-hurley commented on pkg/generator/generator_test.go:
```diff
@@ -545,3 +547,18 @@ func TestGenBuild(t *testing.T) {
+	olmCatalogDirPath := filepath.Join(deployDir, olmCatalogDir, catalogPackageYaml)
+	if err = writeFileAndPrint(olmCatalogDirPath, []byte("sometext"), defaultFileMode); err != nil {
+		t.Error("Test failed. Failed to write file and print:", err)
+	}
```
Can the test also verify that the file was created?

shawn-hurley commented:
Thanks for the PR!

hasbro17 commented:
@AlexNPavel Can you double check why the CI is failing for the openshift cluster. It's failing to set that up.

AlexNPavel commented:
It looks like travis simply failed to download the oc binary. I'll restart the build, see if it works now

AlexNPavel commented:
@hasbro17 the most recent run on travis worked. Seemed to be some temporary network issues

hasbro17 commented on pkg/generator/generator_test.go:
```diff
@@ -545,3 +547,22 @@ func TestGenBuild(t *testing.T) {
+
+	olmCatalogPackagePath := filepath.Join(deployDir, olmCatalogDir, catalogPackageYaml)
+	if err = writeFileAndPrint(olmCatalogPackagePath, []byte("sometext"), defaultFileMode); err != nil {
+		t.Error("Test failed. Failed to write file and print:", err)
```
`t.Error`=>`t.Errorf`

fanminshi replied:
`"Test failed. Failed to write file and print:"` ->
`"failed to write file and print:"`?

also why `t.Error` but not `t.Fatalf`?  I believe `t.Fatalf` exits immediately so that you don't need `return` at line 563.

hasbro17 replied:
Actually yes, t.Errorf() + return can be replaced by t.Fatalf. That will still run all deferred functions.

hasbro17 commented:
LGTM after nit.

fanminshi commented on pkg/generator/generator_test.go:
```diff
@@ -545,3 +547,22 @@ func TestGenBuild(t *testing.T) {
+		return
+	}
+	if _, err := os.Stat(olmCatalogPackagePath); os.IsNotExist(err) {
+		t.Errorf("\nTest failed. Failed to create %s", olmCatalogPackagePath)
```
`t.Errorf("\nTest failed. Failed to create %s", olmCatalogPackagePath)` -> 
`t.Fatalf("failed to create %s", olmCatalogPackagePath)`?

darkowlzz commented:
It felt weird to me as well to use return instead of fatal, but I tried to be consistent with the other tests. I've updated it and added it as part of the previous commit.
Didn't change to `"failed to write file and print:"` because `check_error_case.sh` expects all the `Errorf` and `Fatalf` to start with an uppercase.
Hope it's ready :)

hasbro17 commented:
Thanks for the fix. I've updated the commit message a bit according to our convention.
https://github.com/operator-framework/operator-sdk/blob/master/CONTRIBUTING.MD#format-of-the-commit-message

fanminshi commented:
@darkowlzz not really [check_error_case.sh](https://github.com/operator-framework/operator-sdk/pull/394) ensure the first character is lower cased.
